pvlib/way_faster.py

```diff
@@ -0,0 +1,240 @@
+"""
+Faster ways to calculate single diode model currents and voltages using
+methods from J.W. Bishop (Solar Cells, 1988).
+"""
+
+from collections import OrderedDict
+import numpy as np
+try:
+    from scipy.optimize import brentq, newton
+except ImportError:
+    brentq = NotImplemented
+    newton = NotImplemented
+
+# TODO: update pvsystem.i_from_v and v_from_i to use "gold" method by default
+
+
+def est_voc(photocurrent, saturation_current, nNsVth):
+    """
+    Rough estimate of open circuit voltage useful for bounding searches for
+    ``i`` of ``v`` when using :func:`~pvlib.pvsystem.singlediode`.
```

Comment:
This value is more important as an upper bound on Voc than as an estimate of its value.